### ISSUE TITLE
Persistence of security.php

### ISSUE THREAD

**Issue by ggogel** [adamzammit/limesurvey-docker#21]
The config path `/var/www/html/application/config` contains a file named `security.php`, which holds automatically generated keys that should never change. If this file isn't persisted and the container is recreated, for instance in case of an update, the old `security.php` will be lost and limesurvey will create a new one. This will break the functionality of the instance entirely, as certain values can not be read anymore.

Basically, everybody who is using this deployment right now is on the brink of irrecoverable data loss.

To fix this issue, I mounted a volume into the path  `/var/www/html/application/config`.

Currently, the documentation says nothing about volumes or data persistence in general. Volumes are only defined in the Dockerfile. In my opinion, they should be at least defined in the Docker Compose file.

**Comment by adamzammit** [comment]
Thanks for your note @ggogel 

I have read over the code for LimeSurvey and the contents of security.php are currently fixed. I.e. they are the same for every installation of LimeSurvey unless you change them yourself.

Therefore there is currently only a risk of "irrecoverable data loss" if the user has modified the security.php file inside the container and not kept a copy.

I will make a note in the documentation for this container noting that if you wish to use the encryption feature that you mount a local/volume copy of security.php so there is no risk of losing the encryption keys that you generate.

Alternatively if there is a need/request from the community I would be happy to set up the container to automatically generate new keys (which seems like this should be done instead of everyone using the default which is useless) on new installations otherwise persist existing keys that have been generated.

I will also update the documentation regarding persistence.

Adam

**Comment by adamzammit** [comment]
> Thanks for your note @ggogel
> 
> I have read over the code for LimeSurvey and the contents of security.php are currently fixed. I.e. they are the same for every installation of LimeSurvey unless you change them yourself.
> 
> Therefore there is currently only a risk of "irrecoverable data loss" if the user has modified the security.php file inside the container and not kept a copy.
> 
> I will make a note in the documentation for this container noting that if you wish to use the encryption feature that you mount a local/volume copy of security.php so there is no risk of losing the encryption keys that you generate.
> 
> Alternatively if there is a need/request from the community I would be happy to set up the container to automatically generate new keys (which seems like this should be done instead of everyone using the default which is useless) on new installations otherwise persist existing keys that have been generated.
> 
> I will also update the documentation regarding persistence.
> 
> Adam

My apologies this statement is incorrect.

I will issue a fix as soon as possible.
